cmd/clusterctl/client/cluster/installer.go

```diff
@@ -74,9 +82,50 @@ func (i *providerInstaller) Install() ([]repository.Components, error) {
 
 		ret = append(ret, components)
 	}
+
+	// Verify that Deployments in any components to be installed is available
```

Comment:
I'm checking the deployments status serially instead of in parallel because as I was testing with the parallel version (using WaitGroup and an errChan), it took longer because of the exponential backoff for some of the go routines.

I opted to keep it serially because:
- it took less time on my machine
- the log output from clusterctl was cleaner
- the code looks cleaner as well

I can explain in more detail if y'all would like. 🙂

Reply:
Serially probably makes sense, they're coming up in parallel but we're looking to block until all have completed, so there's no time saving in checking them in parallel